src/main/java/de/tum/cit/aet/artemis/iris/service/pyris/PyrisEventService.java

```diff
@@ -33,33 +32,38 @@ public PyrisEventService(IrisCourseChatSessionService irisCourseChatSessionServi
         this.irisExerciseChatSessionService = irisExerciseChatSessionService;
     }
 
-    /**
-     * Triggers a Pyris pipeline based on the received {@link PyrisEvent}.
-     *
-     * @param event The event object received to trigger the matching pipeline
-     * @throws UnsupportedPyrisEventException if the event is not supported
-     *
-     * @see PyrisEvent
-     */
-    public void trigger(PyrisEvent<? extends AbstractIrisChatSessionService<? extends IrisChatSession>, ?> event) {
-        log.debug("Starting to process event of type: {}", event.getClass().getSimpleName());
+    @EventListener
+    public void handleCompetencyJolSetEvent(CompetencyJolSetEvent event) {
+        log.debug("Processing CompetencyJolSetEvent");
```

Comment:
Could be useful to log the event id for better debugging?

Reply:
Event currently do not have ids. Therefore, it's unfortunately not possible to log the event ids.